tests/__init__.py

```diff
@@ -28,11 +28,10 @@ def module_function(x: int) -> int:
 
 @dataclass
 class AClass:
-    def __init__(self, a: Any, b: Any, c: Any, d: Any = "default_value") -> None:
-        self.a = a
-        self.b = b
-        self.c = c
-        self.d = d
+    a: Any
+    b: Any
+    c: Any
+    d: Any = "default_value"
```

Comment:
dataclass eq before just returned True because it did not see that there are any fields to compare.